tools/flash_shelly.sh

```diff
@@ -0,0 +1,201 @@
+      case $yn in
+        [Yy]* )  echo "Now Flashing.."; curl -F file=shelly-flash.zip http://$1/update; break;;
+        [Nn]* ) echo "Skipping..";break;;
+        * ) echo "Please answer yes or no.";;
+      esac
+    done
+  fi
+  echo "waiting for $1 to reboot"
+  sleep 10
+  if [ $(curl -qsS -m 5 http://$1/rpc/Sys.GetInfo | jq -r .fw_version) == $2 ];then
+    echo "Successfully flashed $1 to $2"
+  else
+    echo "Flash failed!!!"
+  fi
+  rm -f shelly-flash.zip
+  read -p "Press enter to continue"
+}
+
+function probe-info {
+  flash=null
+  device="$2"
+  release_info=$(curl -qsS -m 5 https://api.github.com/repos/mongoose-os-apps/shelly-homekit/releases/latest)
+  lfw=$(echo "$release_info" | jq -r .tag_name)
+
+  if [[ $(curl -qsS -m 5 http://$device/rpc/Shelly.GetInfo) == "Not Found" ]]; then
+    continue
+  fi
+
+  info=$(curl -qsS -m 5 http://$device/rpc/Shelly.GetInfo)
+  cfw=$(echo "$info" | jq -r .version)
+  type=$(echo "$info" | jq -r .app)
+  model=$(echo "$info" | jq -r .model)
+
+  if [[ $model != *Shelly* ]]; then
+    case $type in
+      switch1)
+        model="Shelly1";;
+      switch1pm)
+        model="Shelly1PM";;
+      switch25)
+        model="Shelly25";;
+      shelly-plug-s)
+        model="ShellyPlugS";;
+      *) ;;
+    esac
+  fi
+
+  dlurl=$(echo "$release_info" | jq -r '.assets[] | select(.name=="shelly-homekit-'$model'.zip").browser_download_url')
+
+  if [ $1 == "update" ]; then
+    clear
+    echo "Host: $device"
+    echo "Model: $model"
+    echo "Current: $cfw"
+    echo "Latest: $lfw"
+    echo "DURL: $dlurl"
+
+    cfw_V=$(convert_to_integer $cfw)
+    lfw_V=$(convert_to_integer $lfw)
+
+    if [ $(echo "$lfw_V $cfw_V -p" | dc) == 1 ]; then
+      while true; do
+        read -p "Do you wish to flash $device to firmware version $lfw ? " yn
+        case $yn in
+          [Yy]* )  flash="yes"; break;;
+          [Nn]* ) flash="no";break;;
+          * ) echo "Please answer yes or no.";;
+        esac
+      done
+    else
+      echo "$device DOSE NOT NEED UPDATING..."
+      read -p "Press enter to continue"
+    fi
+
+    if [ "$flash" = "yes" ]; then
+      write_flash $device $lfw $dlurl
+    elif [ "$flash" = "no" ]; then
+      echo "Skipping Flash..."
+      read -p "Press enter to continue"
+    fi
+  else
+    echo "Host: $device"
+    echo "Model: $model"
+    echo "Current: $cfw"
+    echo "Latest: $lfw"
+    echo " "
+  fi
+}
+
+function device-scan {
+  if [ "$2" != "null" ]; then
+    probe-info $1 $2
+  else
+    echo -e '\033[1mScanning for Shelly devices...\033[0m'
+    for device in $(timeout 2 dns-sd -B _hap . | awk '/shelly/ {print $7}'); do
+      probe-info $1 $device.local
+    done
+  fi
+}
+
+function help {
+  echo " -u, --update        Update device(s) to the lastest available firmware."
+  echo " -c, --check-only    Only check for updates."
+  echo " -h, --help          This help text"
+}
+
+if [ -n "$1" ]; then
```

Comment:
use `getopts` to parse command line arguments

Reply:
please check latest script, I did tweak how cmd arguments are handled.